### PR TITLE
Rdm 1893 - Add FixedRadioList validator

### PR DESCRIPTION
**Pull request by mohammedanas**
https://tools.hmcts.net/jira/browse/RDM-1893

As a caseworker
I want to see multiple options in radio button rather than Dropdown
So that I can see all the available options at one go and select one accordingly.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    void baseTypeRegEx() {
+        when(fixedListBaseType.getRegularExpression()).thenReturn("InvalidRegEx");
+        final List<ValidationResult> result = validator.validate("TEST_FIELD_ID",
+                                                                 NODE_FACTORY.textNode("AA"), caseField);
+        assertEquals(1, result.size(), "RegEx validation failed");
+        assertEquals("'AA' failed FixedList Type Regex check: InvalidRegEx", result.get(0).getErrorMessage());
+        assertEquals("TEST_FIELD_ID", result.get(0).getFieldId());
+    }
+
+    private CaseFieldBuilder caseField() {
```
Comment (tzarsmango) [inline src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java]: Would it be better to use TestBuildersUtil.CaseFieldBuilder for code reuse? May be merge `CaseFieldBuilder` class with the one in the TestBuildersUtil class?

File: src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        assertEquals(0, validator.validate("TEST_FIELD_ID", null, null).size(), "Did not catch NULL");
+    }
+
+    @Test
+    void getType() {
+        assertEquals(validator.getType(), BaseType.get("FixedRadioList"), "Type is incorrect");
+    }
+
+    @Test
+    void fieldTypeRegEx() {
```
[the remark this answers is not in the document]
Comment (mohammedanas) [reply src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java]: Removed them

File: src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java (hunk 1 above)
Comment (mohammedanas) [reply src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java]: I have now created a tech debt ticket on this https://tools.hmcts.net/jira/browse/RDM-3366

File: src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        when(fixedListBaseType.getRegularExpression()).thenReturn("InvalidRegEx");
+        final List<ValidationResult> result = validator.validate("TEST_FIELD_ID",
+                                                                 NODE_FACTORY.textNode("AA"), caseField);
+        assertEquals(1, result.size(), "RegEx validation failed");
+        assertEquals("'AA' failed FixedList Type Regex check: InvalidRegEx", result.get(0).getErrorMessage());
+        assertEquals("TEST_FIELD_ID", result.get(0).getFieldId());
+    }
+
+    private CaseFieldBuilder caseField() {
+        return new CaseFieldBuilder(FIELD_ID).withType(FixedListValidator.TYPE_ID)
```
[the remark this answers is not in the document]
Comment (mohammedanas) [reply src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java]: Removed

File: src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+        final List<ValidationResult> result02 = validator.validate("TEST_FIELD_ID", NODE_FACTORY.textNode("BBBBBB"),
+                                                                   caseFieldWithRegEx);
+        assertEquals(1, result02.size(), "BBBBBB failed regular expression check");
+        assertEquals(REGEX_GUIDANCE, result02.get(0).getErrorMessage());
+        assertEquals("TEST_FIELD_ID", result02.get(0).getFieldId());
+    }
+
+    @Test
+    void baseTypeRegEx() {
```
[the remark this answers is not in the document]
Comment (mohammedanas) [reply src/test/java/uk/gov/hmcts/ccd/domain/types/FixedRadioListValidatorTest.java]: Removed

**Pull request merged**